### ISSUE TITLE
Consider adding composerPath to Application

### ISSUE THREAD

**Issue by jradtilbrook** [laravel/framework#22319]
- Laravel Version: 5.*
- PHP Version: Irrelevant
- Database Driver & Version: Irrelevant

### Description:
I have a specific case where I'd like to introduce Laravel as a sub-app to an existing app that is quite old and full of legacy code that doesn't really use any framework. The issue is that the directories used by some core components are already in use and cannot be clobbered or changed.  
I'm thinking of installing Laravel to a subdirectory but keeping the composer files at the root so deploying is simpler (not requiring composer install twice, etc). However there is some code that looks for composer.json or vendor/ by using `base_path()`. It would be really helpful if this could be abstracted like many of the other paths to something like `composer_path()` and allow authors and users to use this instead which can be overridden by users.  
I have currently gotten around this by extending the Application to change the `getNamespace` method and register a custom PackageManifest class. These are the only issues I have seen so far and it is minimal but the fix also seems like a nice feature anyway.  
On a separate note (its not part of this repo; but for brevity), the tinker command also had to be overridden to load the classmap from the correct directory.

Let me know what you think. If its something you would consider, I can make a PR.
Thanks.

### Steps To Reproduce:
Irrelevant

**Comment by laurencei** [comment]
Probably best to post/discuss this at https://github.com/laravel/internals

**Comment by sisve** [comment]
> I'm thinking of installing Laravel to a subdirectory but keeping the composer files at the root so deploying is simpler (not requiring composer install twice, etc).

This sounds really weird. That 150% weird. Are you saying that you want to end up with a /wwwroot/oldcode.php and /wwwroot/laravel/public/index.php? That will probably expose some fun things, like http://domain/laravel/.env and http://domain/laravel/storage/logs/...

Install Laravel to a separate folder and have nginx point /laravel to that folder. This is the proper to do it. It also means that Laravel can be deployed separately, and there is no subdirectory for Laravel. It's two separate applications that just happened to be served by the same nginx instance.

Introducing a framework change because you want to save a call to `composer install` is also slightly weird.

**Comment by jradtilbrook** [comment]
Yeah that's the problem with legacy code-bases. We have made the decision that a rewrite to a completely separate app/repo is too much investment but we want to start using Laravel now alongside our existing code. We can configure the webserver to serve only public/index.php from the subdirectory (ie. no static files - at least for now) and not serve the rest of the laravel/ subdirectory.  
For our case this isn't actually two applications, its just a _newer_ version that now uses Laravel. We also don't want to expose /laravel in the URL.  
I'd argue that its just a nice extra feature that has no impact apart from cases like mine, where the business has refused a rewrite, but developers want the simplicity of a framework like Laravel.  
It's not an issue anyway because I have already configured/overridden the specific files using composer/vendor files so that it works for me. I just thought it would be a nice addition.  
I'll close this since it seems like adoption is not going to happen. Thanks for your time! :smile:
